gpbft/gpbft.go

```diff
@@ -918,6 +948,66 @@ func (i *instance) broadcast(round uint64, step Phase, value ECChain, createTick
 	}
 
 	_ = i.participant.host.RequestBroadcast(mb)
+	i.broadcasted.record(mb)
+}
+
+// tryRebroadcast checks whether the conditions for rebroadcasting messages are
+// met, and if so rebroadcasts messages from current and previous rounds.
+// Rebroadcast is needed when both of the following conditions are met:
+//   - There has been no progress since latest broadcast.
+//   - Rebroadcast backoff duration has elapsed.
+//
+// If there has been progress since latest broadcast this function resets the
+// number of rebroadcast attempts and returns immediately.
+func (i *instance) tryRebroadcast() error {
```

Comment:
This function is now the only one that alters the rebroadcast related parameters. Making all modifications in one place makes the code logic much easier to follow IMO, where `tryRebroadcast` is called explicitly as part of `try<phase>`.